riscv/sim.cc

```diff
@@ -406,10 +406,9 @@ void sim_t::set_rom()
 char* sim_t::addr_to_mem(reg_t paddr) {
   if (!paddr_ok(paddr))
     return NULL;
-  auto desc = bus.find_device(paddr);
+  auto desc = bus.find_device(paddr, 1);
```

Comment:
maybe size 4 is more appropriate? or to include size in `addr_to_mem` ?

Reply:
I used `1` because it preserves the current semantics.  I agree with your second proposal.  When `mmu_t` calls `addr_to_mem`, it should be checking that the entire 4K page surrounding the address is valid, because that is the assumption its TLB makes.